pandas/plotting/_core.py

```diff
@@ -391,14 +393,14 @@ def hist_frame(
 @Appender(_boxplot_doc)
 def boxplot(
     data,
-    column=None,
+    column: Optional[Union[str, List[str]]] = None,
     by=None,
     ax=None,
-    fontsize=None,
-    rot=0,
-    grid=True,
-    figsize=None,
-    layout=None,
+    fontsize: Optional[Union[float, str]] = None,
+    rot: Optional[Union[float, int]] = 0,
```

Comment:
```suggestion
    rot: Optional[float] = 0,
```
int is a subtype of float